### ISSUE TITLE
Going To Camp support

### ISSUE THREAD

**Issue by acaloiaro** [juftin/camply#94]
**Is your feature request related to a problem? Please describe.**
This is a tracking issue for adding [Going To Camp](https://goingtocamp.com/) support. 

A Going to Camp provider will add support for the following areas

- Washing State Parks
- Wisconsin State Parks
- City of Oroville (WA)
- Tacoma Power Parks (WA)
- Longpoint Area (CA)

- Saugeen Valley Conservation Area (Canada) 
- St. Clair Conservation Authority (Canada)
- Yukon Parks (Canada) 

Other Going to Camp providers may be possible, but it looks like they use some sort of self-hosted solution that makes the API more difficult to work with (and sometimes blocked because javascript is not active, or an unacceptable browser UA is in use).

---

I plan to implement an initial proof of concept shortly. 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by SoccerCore11** [comment]
What would it take to get the https://midnrreservations.com/ site working with this also? Looks like they are using the self-hosted solution of Going to Camp / Camis solution as you described.

**Comment by acaloiaro** [comment]
@SoccerCore11 Good question to which I don't know the answer. GoingToCamp support is coming in the `0.8.0` camply release and I haven't looked into adding the self-hosted versions of camply. 

It takes a fair amount of effort to reverse engineer the URL structure of the API, so I don't expect to look into it in the near future. However, I do imagine it's doable.

**Comment by acaloiaro** [comment]
I looked into it briefly. I think this will work trivially without much code change. Changes needed are:

1. Add a user agent that looks like a real browser. The reservation site is serving up a captcha instead of the necessary JSON results. I used the agent served by my browser (firefox), and it allowed my `curl` request with correct results. 
2. A `RecreationArea` entry needs to be added to the recreation area list: https://github.com/juftin/camply/pull/170/files#diff-27cb2d1da317d1840aeb797e9a89748b2fd5bf43e193de78b9821216b342825dR24

From my original post:

> Other Going to Camp providers may be possible, but it looks like they use some sort of self-hosted solution that makes the API more difficult to work with (**and sometimes blocked because javascript is not active, or an unacceptable browser UA is in use**).

This is probably true for many or all recreation areas. It's possible that the self-hosted version of going to camp blocks non-standard UAs by default.
